File: cpp/1_twoSum.cpp
# include <vector>
# include <map>


using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      std::map<int, int> num2idx;
      for (int i = 0; i < nums.size(); i++) {
          int num = nums[i];
          num2idx.insert({num, i});
      } 
      for (int i = 0; i < nums.size(); i++) {
          int curr_num = target - nums[i];
          if (num2idx.find(curr_num) != num2idx.end() && num2idx[curr_num] != i) {
              return std::vector<int>({i, num2idx[curr_num]});
          }
      }
      return std::vector<int>({});
    }
};

int main() {
    Solution solution;
    std::vector<int> nums = {2,7,11,15};
    int target = 9;
    std::vector<int> res = solution.twoSum(nums, target);

    return 0;
}

File: wangtiantian/python/1_twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num2idx = {}
        for i, num in enumerate(nums):
            num2idx[num] = i

        for i, num in enumerate(nums):
            remain = target - num 
            if remain in num2idx and num2idx[remain] !=  i:
                return [i, num2idx[remain]]